api/static/app/main.js

```diff
@@ -505,12 +505,19 @@ if (!nobrowsercheck && !isChrome()) {
     // Periodically load data
     var REFRESH_TIME_MINUTES = 5;
     function fetchAndReschedule() {
+        // If data doesn't load in 30 secs, show connection warning
+        timeout_interval = setTimeout(function(){
+            document.getElementById('connection-warning').style.display = "block";
+        }, 30000);
         queue()
             .defer(d3.json, 'europe.topo.json')
             .defer(d3.json, ENDPOINT + '/v1/production')
             .defer(d3.json, ENDPOINT + '/v1/solar')
             .defer(d3.json, ENDPOINT + '/v1/wind')
-            .await(dataLoaded);
+            .await(function(){
+                clearInterval(timeout_interval);
```

Comment:
If it takes more than 30sec, then warning is displayed. The warning element has an animation, which is triggered when the element is first shown. This animation shows the element and hides it 15sec later.

I've found a problem, however: only on the first timeout the warning will show, because the animation only starts when the element goes from `display: none` to `display: block`.

I've added a fix - see next commit.



The only problem I see is: The animation will only show once.. for 15sec.